### PR TITLE
Deprecate MTQ

### PR DESCRIPTION
**Pull request by Barakmor1**
There are several problems with the MTQ Operator:

Namespace Locking:
- While it functions, bugginess has been observed in the wild.

Tightly coupled with KubeVirt Code:
- One-to-one changes are required for updates to logic in either repository.

Modifies the quota set by the cluster admin.

Accounting nuances:
- Both source and target pods are counted until migration completes.

The AAQ Operator resolves all these issues;therefore, the MTQ should be dropped, and we should fully commit to the AAQ Operator.

Deprecation Plan:

1. Deprecate the MTQ flag, disabling it even if activated.

2. Remove MTQ from CSVs and eliminate all MTQ-related code from HCO.

This commit accomplishes the first step.The reason for the two-step removal is the concern regarding updates with MTQ enabled.

In that case, the MTQ Operator won't be available to remove MTQ leftovers.

**What this PR does / why we need it**:

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly

**Jira Ticket**:

```jira-ticket
https://issues.redhat.com/browse/CNV-40849
```

**Release note**:

```release-note
Deprecate MTQ
```

### CODE REVIEW

File: docs/cluster-configuration.md [hunk 1]
```diff
@@ -206,12 +206,6 @@ VMI example:
 
 **Default**: `false`
 
-### enableManagedTenantQuota Feature Gate
-If set to true, enables the Managed Tenant Quota (MTQ) feature. See more details 
-[here](https://github.com/kubevirt/managed-tenant-quota).
-
-**Default**: `false`
-
```
Comment (nunnatsa) [inline docs/cluster-configuration.md:214]: Please don't delete, but just say it's deprecated and ignored

Comment (Barakmor1) [reply docs/cluster-configuration.md]: Done

File: docs/cluster-configuration.md [hunk 2]
```diff
@@ -207,9 +207,11 @@ VMI example:
 **Default**: `false`
 
 ### enableManagedTenantQuota Feature Gate
-If set to true, enables the Managed Tenant Quota (MTQ) feature. See more details 
+If set to true, enables the Managed Tenant Quota (MTQ) feature. See more details
 [here](https://github.com/kubevirt/managed-tenant-quota).
```
Comment (nunnatsa) [inline docs/cluster-configuration.md:211]: I think we can drop these two lines. Maybe also the "default" line - we don't care about it now, I think. 

Please just leave the line about deprecation.

**Comment by nunnatsa** [comment]
hco-e2e-upgrade-prev-operator-sdk-aws lane passed
/override ci/prow/hco-e2e-upgrade-prev-operator-sdk-azure

**Comment by nunnatsa** [comment]
/override coverage/coveralls

Will fix in following PR

**Comment by nunnatsa** [comment]
okd CI is still out of order, let's ignore it
/override ci/prow/okd-hco-e2e-operator-sdk-aws
/override ci/prow/okd-hco-e2e-operator-sdk-gcp
/override ci/prow/okd-hco-e2e-upgrade-operator-sdk-aws
/override ci/prow/okd-hco-e2e-upgrade-operator-sdk-gcp

**Review by nunnatsa — APPROVED** [review]
/approve

**Comment by nunnatsa** [comment]
OKD lanes are curretnly broken. 
/override ci/prow/okd-hco-e2e-operator-sdk-aws
/override ci/prow/okd-hco-e2e-operator-sdk-gcp
/override ci/prow/okd-hco-e2e-upgrade-operator-sdk-aws
/override ci/prow/okd-hco-e2e-upgrade-operator-sdk-gcp

**Pull request merged**